modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,407 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background for that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since 1.0.0
+ */
+
+/**
+ * Add Dominant Color preload support to WordPress.
+ *
+ * @since 1.0.0
+ */
+class wp_Dominant_Color {
+
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+
+		add_filter( 'wp_print_scripts', array( $this, 'add_styles' ) );
+
+		add_filter( 'wp_generate_attachment_metadata', array( $this, 'dominant_color_metadata' ), 10, 2 );
+		add_filter( 'wp_generate_attachment_metadata', array( $this, 'has_transparency_metadata' ), 10, 2 );
+
+		// do we have the new filter or are duplicating core the functions?
+		if ( has_filter( 'wp_content_img_tag' ) ) {
+			add_filter( 'wp_content_img_tag', array( $this, 'tag_add_adjust' ), 20, 3 );
+		} else {
+			add_filter( 'the_content', array( $this, 'filter_content_tags' ), 20 );
+			add_filter( 'the_excerpt', array( $this, 'filter_content_tags' ), 20 );
+			add_filter( 'widget_text_content ', array( $this, 'filter_content_tags' ), 20 );
+			add_filter( 'widget_block_content', array( $this, 'filter_content_tags' ), 20 );
+			add_filter( 'wp_dominant_color_img_tag_add_adjust', array( $this, 'tag_add_adjust' ), 10, 3 );
+		}
```

Comment:
```suggestion
	add_filter( 'wp_content_img_tag', array( $this, 'tag_add_adjust' ), 20, 3 );
	add_filter( 'wp_dominant_color_img_tag_add_adjust', array( $this, 'tag_add_adjust' ), 10, 3 );
	$filters = array( 'the_content', 'the_excerpt', 'widget_text_content', 'widget_block_content' );
	foreach( $filters as $filter ) {
	       add_filter( $filter, array( $this, 'filter_content_tags' ), 20 );
	}	
```

Let's do this, it's cleaner.